build_test.go

```diff
@@ -1343,168 +1275,285 @@ func testBuild(t *testing.T, when spec.G, it spec.S) {
 		})
 
 		when("Publish option", func() {
-			when("true", func() {
-				var remoteRunImage *fakes.Image
+			var remoteRunImage, builderImageSupportingCreator, fakeLifecycleImage *fakes.Image
 
-				it.Before(func() {
-					remoteRunImage = fakes.NewImage("default/run", "", nil)
-					h.AssertNil(t, remoteRunImage.SetLabel("io.buildpacks.stack.id", defaultBuilderStackID))
-					h.AssertNil(t, remoteRunImage.SetLabel("io.buildpacks.stack.mixins", `["mixinA", "mixinX", "run:mixinZ"]`))
-					fakeImageFetcher.RemoteImages[remoteRunImage.Name()] = remoteRunImage
-				})
+			it.Before(func() {
+				remoteRunImage = fakes.NewImage("default/run", "", nil)
+				h.AssertNil(t, remoteRunImage.SetLabel("io.buildpacks.stack.id", defaultBuilderStackID))
+				h.AssertNil(t, remoteRunImage.SetLabel("io.buildpacks.stack.mixins", `["mixinA", "mixinX", "run:mixinZ"]`))
+				fakeImageFetcher.RemoteImages[remoteRunImage.Name()] = remoteRunImage
+
+				lifecycleVersionSupportingCreator := "0.7.5"
+
+				builderImageSupportingCreator = newFakeBuilderImage(
+					t,
+					tmpDir,
+					"example.com/supportscreator/builder:tag",
+					"some.stack.id",
+					lifecycleVersionSupportingCreator,
+				)
+				h.AssertNil(t, builderImageSupportingCreator.SetLabel("io.buildpacks.stack.mixins", `["mixinA", "build:mixinB", "mixinX", "build:mixinY"]`))
+				fakeImageFetcher.LocalImages[builderImageSupportingCreator.Name()] = builderImageSupportingCreator
 
-				it.After(func() {
-					remoteRunImage.Cleanup()
-				})
+				fakeLifecycleImage = fakes.NewImage(fmt.Sprintf("%s:%s", lifecycleImageRepo, lifecycleVersionSupportingCreator), "", nil)
+				fakeImageFetcher.LocalImages[fakeLifecycleImage.Name()] = fakeLifecycleImage
+			})
 
+			it.After(func() {
+				remoteRunImage.Cleanup()
+				builderImageSupportingCreator.Cleanup()
+				fakeLifecycleImage.Cleanup()
+			})
+
+			when("true", func() {
 				it("uses a remote run image", func() {
 					h.AssertNil(t, subject.Build(context.TODO(), BuildOptions{
 						Image:   "some/app",
-						Builder: builderName,
+						Builder: defaultBuilderName,
 						Publish: true,
 					}))
 					h.AssertEq(t, fakeLifecycle.Opts.Publish, true)
 
 					args := fakeImageFetcher.FetchCalls["default/run"]
 					h.AssertEq(t, args.Daemon, false)
 
-					args = fakeImageFetcher.FetchCalls[builderName]
+					args = fakeImageFetcher.FetchCalls[defaultBuilderName]
 					h.AssertEq(t, args.Daemon, true)
 				})
 
-				when("false", func() {
-					it("uses a local run image", func() {
-						h.AssertNil(t, subject.Build(context.TODO(), BuildOptions{
-							Image:   "some/app",
-							Builder: builderName,
-							Publish: false,
-						}))
-						h.AssertEq(t, fakeLifecycle.Opts.Publish, false)
+				when("builder is untrusted", func() {
```

Comment:
The tests in this section hurt my brain and were the trickiest to get right. They could probably benefit from an extra careful review :)